compiler/noirc_evaluator/src/acir/tests/arrays.rs

```diff
@@ -382,15 +382,42 @@ fn non_homogenous_array_dynamic_access() {
     ASSERT w41 = 21
     ASSERT w42 = 22
     ASSERT w43 = 25
-    INIT b1 = [w32, w33, w34, w35, w36, w37, w38, w39, w40, w41, w42, w43]
-    READ w44 = b1[w28]
-    READ w45 = b0[w44]
-    ASSERT w46 = w44 + 1
-    READ w47 = b0[w46]
-    ASSERT w48 = w46 + 1
-    READ w49 = b0[w48]
-    ASSERT w29 = w45
-    ASSERT w30 = w47
-    ASSERT w31 = w49
+    ASSERT w44 = 28
+    ASSERT w45 = 29
+    ASSERT w46 = 32
+    ASSERT w47 = 35
+    ASSERT w48 = 36
+    ASSERT w49 = 39
+    ASSERT w50 = 42
+    ASSERT w51 = 43
+    ASSERT w52 = 46
+    ASSERT w53 = 49
+    ASSERT w54 = 50
+    ASSERT w55 = 53
+    ASSERT w56 = 56
+    ASSERT w57 = 57
+    ASSERT w58 = 60
+    ASSERT w59 = 63
+    ASSERT w60 = 64
+    ASSERT w61 = 67
+    ASSERT w62 = 70
+    ASSERT w63 = 71
+    ASSERT w64 = 74
+    ASSERT w65 = 77
+    ASSERT w66 = 78
+    ASSERT w67 = 81
+    ASSERT w68 = 84
+    ASSERT w69 = 85
+    ASSERT w70 = 88
+    INIT b1 = [w32, w33, w34, w35, w36, w37, w38, w39, w40, w41, w42, w43, w44, w45, w46, w47, w48, w49, w50, w51, w52, w53, w54, w55, w56, w57, w58, w59, w60, w61, w62, w63, w64, w65, w66, w67, w68, w69, w70]
```

Comment:
This looks like more than one additional element. I would expect us to got from 12 values to 15 values.